src/main/java/com/coveo/pushapiclient/UserAgent.java

```diff
@@ -0,0 +1,22 @@
+package com.coveo.pushapiclient;
+
+public enum UserAgent {
+  SAP_COMMERCE_CLOUD_V1 {
+    @Override
+    public String toString() {
+      return "SAPCommerceCloud/v1";
+    }
+  },
+  SAP_COMMERCE_CLOUD_V2 {
+    @Override
+    public String toString() {
+      return "SAPCommerceCloud/v2";
+    }
+  },
+  SAP_COMMERCE_CLOUD_V3 {
+    @Override
+    public String toString() {
+      return "SAPCommerceCloud/v3";
+    }
+  }
+}
```

Comment:
I agree with Sami that SAP references are a leak of concerns.
We have precedents for similar open traps, such as the `versions` field in Relay to report what uses Relay.
What we did instead is normalize what are the expected strings for Coveo products.
So I'd vote in favor of removing the enum and going with a String.
The value themselves should be moved upward in the supply chain.

Reply:
> a client implementing with the SDK could enter any old junk

Wouldn't a client entering an unrecognized value be better to them picking an enum value, as it would give a false impression of CoSAP usage? A string is in a sense hiding how our dashboard might be working.

> As for the versions, when you say connector, I assume you mean the CoSAP connector in Hybris?

Yep, atm we're only tracking major versions, so updating the enum would be less frequent. It could be useful to know full versions though, which would make updates to the enum more frequent. Considering the userAgent defaults to a full Java SDK version, it feels consistent to have the full version. The point is more philosophical though, that the enum is coupling two packages together

Another philosophical angle is if we were supporting SDKs in multiple languages, one would expect the SDKs to behave similarly. However, we would not reference SAP in a python or nodejs SDK? So having it in the Java one feels like a sign that we're blurring the boundaries,

Reply:
Agreed with Louis that a `string[]` (or a string concatenating the values in the array) could be interesting too. Would that complexify the dashboard logic too much though?

Reply:
If we go for a string then, do we want to allow the user to enter anything they want? Looking at the [User-Agent Spec](https://www.rfc-editor.org/rfc/rfc9110.html#name-user-agent) the [comment](https://www.rfc-editor.org/rfc/rfc9110.html#comments) allows pretty much most stuff. Or should we think about constraining it to reject special chars? What approach was taken before?

Reply:
The `source` value on events is [restricted](https://github.com/coveo-platform/analytics_schema/blob/master/schemas/common/types.json#L281) to an array of strings with format `<name>@<version>`. That said, we could keep it a free string if this is too restrictive,

Reply:
use of a string array has been introduced and the user agents are now concatenated into the header value